core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -668,7 +668,7 @@ public DiscoveryNodes build() {
             ImmutableOpenMap.Builder<String, DiscoveryNode> masterNodesBuilder = ImmutableOpenMap.builder();
             ImmutableOpenMap.Builder<String, DiscoveryNode> ingestNodesBuilder = ImmutableOpenMap.builder();
             Version minNodeVersion = Version.CURRENT;
-            Version maxNodeVersion = Version.CURRENT;
+            Version maxNodeVersion = null;
```

Comment:
I think I'm confused why there's an asymmetry between min and max, would you please explain (or fix)?